src/js/load-package.ts

```diff
@@ -354,104 +434,75 @@ export async function loadPackage(
-    if (loaded === uri || uri === DEFAULT_CHANNEL) {
+    if (
+      loaded === pkg_metadata.channel ||
+      pkg_metadata.channel === DEFAULT_CHANNEL
+    ) {
       messageCallback(`${pkg} already loaded from ${loaded}`);
     } else {
       errorCallback(
-        `URI mismatch, attempting to load package ${pkg} from ${uri} ` +
+        `URI mismatch, attempting to load package ${pkg} from ${pkg_metadata.channel} ` +
           `while it is already loaded from ${loaded}. To override a dependency, ` +
           `load the custom package first.`
       );
     }
   }
 
-  if (toLoad.size === 0 && toLoadShared.size === 0) {
+  if (toLoad.size === 0) {
     messageCallback("No new packages to load");
     return;
   }
 
-  const packageNames = [...toLoad.keys(), ...toLoadShared.keys()].join(", ");
+  const packageNames = [...toLoad.keys()].join(", ");
+  const loaded = new Set<string>();
+  const failed = new Map<string, Error>();
   const releaseLock = await acquirePackageLock();
   try {
     messageCallback(`Loading ${packageNames}`);
-    const sharedLibraryLoadPromises: { [name: string]: Promise<Uint8Array> } =
-      {};
-    const packageLoadPromises: { [name: string]: Promise<Uint8Array> } = {};
-    for (const [name, channel] of toLoadShared) {
-      if (loadedPackages[name]) {
-        // Handle the race condition where the package was loaded between when
-        // we did dependency resolution and when we acquired the lock.
-        toLoadShared.delete(name);
-        continue;
-      }
-      sharedLibraryLoadPromises[name] = downloadPackage(name, channel);
-    }
-    for (const [name, channel] of toLoad) {
+    for (const [name] of toLoad) {
       if (loadedPackages[name]) {
         // Handle the race condition where the package was loaded between when
         // we did dependency resolution and when we acquired the lock.
         toLoad.delete(name);
         continue;
       }
-      packageLoadPromises[name] = downloadPackage(name, channel);
-    }
 
-    const loaded: string[] = [];
-    const failed: { [name: string]: any } = {};
-    // TODO: add support for prefetching modules by awaiting on a promise right
-    // here which resolves in loadPyodide when the bootstrap is done.
-    const sharedLibraryInstallPromises: { [name: string]: Promise<void> } = {};
-    const packageInstallPromises: { [name: string]: Promise<void> } = {};
-    for (const [name, channel] of toLoadShared) {
-      sharedLibraryInstallPromises[name] = sharedLibraryLoadPromises[name]
-        .then(async (buffer) => {
-          await installPackage(name, buffer, channel);
-          loaded.push(name);
-          loadedPackages[name] = channel;
-        })
-        .catch((err) => {
-          console.warn(err);
-          failed[name] = err;
-        });
+      // TODO: add support for prefetching modules by awaiting on a promise right
+      // here which resolves in loadPyodide when the bootstrap is done.
```

Comment:
I don't understand this comment.

Reply:
Well that comment is originally written by you: https://github.com/pyodide/pyodide/blame/939b615df00b1dd46d463f22c5fcaf01bd483da2/src/js/load-package.ts#L412

I guess you wanted to add a feature to prefetch modules when loadPyodide is called?

Reply:
Ah okay I understand what I meant now. That probably would be good to do someday.